docs/content/concepts/metadata-tags/asset-metadata/table-metadata.mdx

```diff
@@ -0,0 +1,192 @@
+            columns=[
+                TableColumn(
+                    "name",
+                    "string",
+                    description="The name of the person",
+                ),
+                TableColumn(
+                    "age",
+                    "int",
+                    description="The age of the person",
+                ),
+            ]
+        )
+    },
+)
+def my_asset(): ...
+
+
+# Here, the schema isn't known until runtime
+@asset(deps=[AssetKey("source_bar"), AssetKey("source_baz")])
+def my_other_asset():
+    column_names = ...
+    column_types = ...
+
+    columns = [
+        TableColumn(name, column_type)
+        for name, column_type in zip(column_names, column_types)
+    ]
+
+    yield MaterializeResult(
+        metadata={"dagster/column_schema": TableSchema(columns=columns)}
+    )
+```
+
+The schema for `my_asset` will be visible in the Dagster UI. You may optionally attach <PyObject object="TableColumnConstraints"/> to each column to provide additional context about the values in the column:
+
+```python file=/concepts/metadata-tags/asset_column_schema_constraints.py
+from dagster import (
+    AssetKey,
+    MaterializeResult,
+    TableColumn,
+    TableColumnConstraints,
+    TableSchema,
+    asset,
+)
+
+
+@asset(
+    deps=[AssetKey("source_bar"), AssetKey("source_baz")],
+    metadata={
+        "dagster/column_schema": TableSchema(
+            columns=[
+                TableColumn(
+                    "name",
+                    "string",
+                    description="The name of the person",
+                ),
+                TableColumn(
+                    "age",
+                    "int",
+                    description="The age of the person",
+                    constraints=TableColumnConstraints(nullable=False, other=[">0"]),
+                ),
+            ]
+        )
+    },
+)
+def my_asset(): ...
+```
+
+### For assets loaded from integrations
+
+Column schemas are currently supported in the dbt integration. Refer to the [dbt documentation](/integrations/dbt/reference) for more information.
+
+---
+
+## Attaching row count
+
+### For assets defined in Dagster
+
+<Image
+alt="Row count for an asset in the Dagster UI"
+src="/images/concepts/metadata-tags/metadata-row-count.png"
+width={1921}
+height={559}
+/>
+
+You may attach row count schema metadata to Dagster assets as [materialization metadata](/concepts/metadata-tags/asset-metadata#attaching-materialization-metadata) to provide additional context about the number of rows in a materialized table. Dagster will let you track changes in the row count over time, and you can use this information to monitor data quality.
+
+To attach schema metadata to an asset, you will need to attach a numerical value to the `dagster/row_count` key in the metadata parameter of the <PyObject object="MaterializeResult" /> object returned by the asset function.
+
+Below is an example of how to attach row count metadata to an asset:
```

Comment:
I don't think you need this sentence. You could use the previous paragraph to lead into this, ex:

> To attach schema metadata to an asset, you will need to attach a numerical value to the `dagster/row_count` key in the metadata parameter of the <PyObject object="MaterializeResult" /> object returned by the asset function. For example: